utils/chartUtils.js

```diff
@@ -0,0 +1,83 @@
+import { emotionData } from "@/lib/db";
+
+export const chartPresets = {
+  tension: {
+    title: "Tension Chart",
+    xTitle: "time",
+    yTitle: "tension",
+    scatter: true,
+  },
+  emotionShares: {
+    title: "Emotion Shares",
+    xTitle: "emotion",
+    yTitle: "percentage",
+    scatter: false,
+  },
+  emotionIntensity: {
+    title: "Average Intensities of Emotions",
+    xTitle: "emotion",
+    yTitle: "average intensity",
+    scatter: false,
+  },
+};
+
+export function countEmotions(entries) {
+  const emotionsToCount = emotionData.map((element) => element.name);
+  function add(a, b) {
+    return Number(a) + Number(b);
+  }
+  function getAverageValue(entries) {
+    const averageValue =
+      entries.length === 0
+        ? 0
+        : Math.round(
+            entries.map((entry) => entry.intensity).reduce(add, 0) /
+              entries.length
+          );
+    return averageValue;
+  }
+  const preResults = emotionsToCount.map((element) => {
+    const foundEntries = entries.filter((entry) => entry.emotion === element);
+    const object = {
+      emotion: element,
+      count: foundEntries.length,
+      averageIntensity: getAverageValue(foundEntries),
+      foundEntries,
+    };
+    return object;
+  });
+  const entriesSum = preResults.map((result) => result.count).reduce(add, 0);
+  const countResults = preResults.map((result) => {
+    const percentage = Math.round((result.count / entriesSum) * 100);
+    const object = {
+      ...result,
+      percentage,
+    };
+    return object;
+  });
+  return countResults;
+}
+
+export function doTensionChartData(entries) {
```

Comment:
As all data is somehow calculated one could argue that one name is as good as the other ... or that mine was better because shorter and still descriptive ;) but I changed it anyways